src/ch/ch_conf.c

```diff
@@ -0,0 +1,235 @@
+                                         0,
+                                         NULL)) == NULL)
+        goto cleanup;
+
+    if (virCapabilitiesAddGuestDomain(guest,
+                                      VIR_DOMAIN_VIRT_CH,
+                                      NULL,
+                                      NULL,
+                                      0,
+                                      NULL) == NULL)
+        goto cleanup;
+
+    return caps;
+
+cleanup:
+    virObjectUnref(caps);
+    return NULL;
+}
+
+/**
+ * virCHDriverGetCapabilities:
+ *
+ * Get a reference to the virCapsPtr instance for the
+ * driver. If @refresh is true, the capabilities will be
+ * rebuilt first
+ *
+ * The caller must release the reference with virObjetUnref
+ *
+ * Returns: a reference to a virCapsPtr instance or NULL
+ */
+virCapsPtr virCHDriverGetCapabilities(virCHDriverPtr driver,
+                                      bool refresh)
+{
+    virCapsPtr ret;
+    if (refresh)
+    {
+        virCapsPtr caps = NULL;
+        if ((caps = virCHDriverCapsInit()) == NULL)
+            return NULL;
+
+        chDriverLock(driver);
+        virObjectUnref(driver->caps);
+        driver->caps = caps;
+    }
+    else
+    {
+        chDriverLock(driver);
+    }
+
+    ret = virObjectRef(driver->caps);
+    chDriverUnlock(driver);
+    return ret;
+}
+
+virDomainXMLOptionPtr
+chDomainXMLConfInit(virCHDriverPtr driver)
+{
+    virCHDriverDomainDefParserConfig.priv = driver;
+    return virDomainXMLOptionNew(&virCHDriverDomainDefParserConfig,
+                                 &virCHDriverPrivateDataCallbacks,
+                                 NULL, NULL, NULL);
+}
+
+virCHDriverConfigPtr
+virCHDriverConfigNew(void)
+{
+    virCHDriverConfigPtr cfg;
+
+    if (virCHConfigInitialize() < 0)
+        return NULL;
+
+    if (!(cfg = virObjectNew(virCHDriverConfigClass)))
+        return NULL;
+
+    cfg->stateDir = g_strdup(CH_STATE_DIR);
+    cfg->logDir = g_strdup(CH_LOG_DIR);
+
+    return cfg;
+}
+
+virCHDriverConfigPtr virCHDriverGetConfig(virCHDriverPtr driver)
+{
+    virCHDriverConfigPtr cfg;
+    chDriverLock(driver);
+    cfg = virObjectRef(driver->config);
+    chDriverUnlock(driver);
+    return cfg;
+}
+
+static void
+virCHDriverConfigDispose(void *obj)
+{
+    virCHDriverConfigPtr cfg = obj;
+
+    VIR_FREE(cfg->stateDir);
+    VIR_FREE(cfg->logDir);
+}
+
+static int
+chExtractVersionInfo(int *retversion)
+{
+    int ret = -1;
+    unsigned long version;
+    char *help = NULL;
+    char *tmp;
+    virCommandPtr cmd = virCommandNewArgList(CH_CMD, "--version", NULL);
+
+    if (retversion)
+        *retversion = 0;
+
+    virCommandAddEnvString(cmd, "LC_ALL=C");
+    virCommandSetOutputBuffer(cmd, &help);
+
+    if (virCommandRun(cmd, NULL) < 0)
+        goto cleanup;
+
+    tmp = help;
+
+    /* expected format: cloud-hypervisor v<major>.<minor>.<micro> */
+    if ((tmp = STRSKIP(tmp, "cloud-hypervisor v")) == NULL)
```

Comment:
Good to know, though that seems like maybe a build process related instead of version looking at the code. @sboeuf any feedback here because as far as I can tell https://github.com/cloud-hypervisor/cloud-hypervisor/commit/2ed17abb5c44f1ae00dd098c9f50e2ac572399c5 introduced the version output handling though perhaps it is a clap related change.

Reply:
The commit https://github.com/cloud-hypervisor/cloud-hypervisor/commit/ebd83699dcfc0bf9df425a1c619462266411884d introduced the version support, therefore I agree we can limit the CH's minimal version.
Actually, talking about limiting the version, I believe it'd be easier to use the latest version (0.9.0) as the minimum version since we're developing this libvirt driver based on this. And since we didn't reach `1.0` yet, we might have some breaking changes (regarding the API/CLI) between different versions. So I'd suggest that we don't try to support too many versions for now.